packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+			.should("have.attr", "value-state", "Negative");
+	});
+
+	it("Custom Validation None", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/09/2020 - 10/10/2020")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value-state", "None");
+	});
+
+	it("Selected dates are updated after value update in the input field", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		const timestamp_9_Sep_2020 = 1599609600;
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/09/2020 - 10/10/2020")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-daypicker")
+			.invoke("attr", "timestamp")
+			.should("eq", timestamp_9_Sep_2020.toString());
+	});
+
+	it("Is delimiter set", () => {
+		cy.mount(<DateRangePicker formatPattern="MMM d, y" delimiter="@"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("Feb 25, 2022 @ Feb 28, 2022")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.invoke("attr", "delimiter", "###")
+			.should("have.attr", "delimiter", "###");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.should("have.attr", "value", "Feb 25, 2022 ### Feb 28, 2022");
+	});
+
+	it("startDateValue and endDateValue getter", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("27/09/2019 - 10/10/2019")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.then($datePicker => {
+				cy.wrap({
+					startDateValue: () => $datePicker[0].startDateValue,
+					endDateValue: () => $datePicker[0].endDateValue
+				})
+					.as("dates")
+					.invoke("startDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp)).to.deep.equal(new Date(2019, 8, 27));
+					});
+
+				cy.get("@dates")
+					.invoke("endDateValue")
+					.then(date => {
+						const timestamp = date ? date.getTime() : new Date().getTime();
+						expect(new Date(timestamp as number)).to.deep.equal(new Date(2019, 9, 10));
+					});
+			});
+	});
+
+	it("Setting the same date for first & last is possible", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type